elastic/kibana
# Add Kibana version header when requesting Synthetics Service's /allowed endpoint

## Summary

This PR makes Kibana include a `X-kibana-version` header in its request to the Synthetics Service's `/allowed` endpoint.

The service can use this header to decide whether to allow Kibana in.

### Implementation Details

I have previously elaborated on Slack about whether we should do version detection through including any extra information on certificates or by having the service check the requesting cluster's information directly.

However, changing the certificates or giving the service access to the clusters would be either complicated or could easily make things insecure. Therefore, given we do this check only for Kibana UI purposes (and the only route which checks this is the actual `/allowed` route), I decided to add a header.

> ⚠️ I've also removed an unnecessary check on `this.authorization` given we already have a top-level check for it which will always return `true` for whether the user is allowed when using basic auth.


### Checklist

Delete any items that are not applicable to this PR.



### For maintainers


## How to test this PR

1. [Install and run `mitmproxy`](https://mitmproxy.org/) so that you can intercept requests on behalf of the server (unless you want to run the server and add `fmt.Println` yourself).
    TIP: To start `mitmproxy` just run `mitmweb --listen-port 10001` after installing it.
    PS: You must ensure Kibana is configured to talk to the service on port `10001` of `localhost` by setting it on `xpack.uptime.service.devUrl`.
2. Ensure your Kibana is configured not to use `basic` auth by removing the basic auth fields `xpack.uptime.service.username` and `xpack.uptime.service.password`.
3. Checkout this branch and start Kibana
4. Allow Kibana a few minutes to do requests to the `/allowed` endpoint of the service
5. You should see headers in the requests intercepted by the `mitmproxy` as shown below

![Screenshot 2022-07-25 at 14 12 54](https://user-images.githubusercontent.com/6868147/180785686-57741bdd-69cd-4b2f-b238-7648245d6df2.png)

## Testing on the service side

* Please ensure that we still disallow users whose "enabled" flag is explicitly set to false
